cachi2/core/package_managers/yarn_classic/main.py

```diff
@@ -1,4 +1,69 @@
-from cachi2.core.models.output import EnvironmentVariable
+import logging
+
+import semver
+
+from cachi2.core.errors import PackageManagerError
+from cachi2.core.models.input import Request
+from cachi2.core.models.output import Component, EnvironmentVariable, RequestOutput
+from cachi2.core.package_managers.yarn.utils import run_yarn_cmd
+from cachi2.core.rooted_path import RootedPath
+
+log = logging.getLogger(__name__)
+
+
+MIRROR_DIR = "deps/yarn-classic"
+
+
+def fetch_yarn_source(request: Request) -> RequestOutput:
+    """Process all the yarn source directories in a request."""
+    components: list[Component] = []
+
+    def _ensure_mirror_dir_exists(output_dir: RootedPath) -> None:
+        output_dir.join_within_root(MIRROR_DIR).path.mkdir(parents=True, exist_ok=True)
+
+    for package in request.yarn_classic_packages:
+        path = request.source_dir.join_within_root(package.path)
+        _ensure_mirror_dir_exists(request.output_dir)
+        prefetch_env = _get_prefetch_environment_variables(request.output_dir)
+        _verify_corepack_yarn_version(path, prefetch_env)
+        _fetch_dependencies(path, prefetch_env)
+
+    return RequestOutput.from_obj_list(
+        components, _generate_build_environment_variables(), project_files=[]
+    )
+
+
+def _fetch_dependencies(source_dir: RootedPath, env: dict[str, str]) -> None:
+    """Fetch dependencies using 'yarn install'.
+
+    :param source_dir: the directory in which the yarn command will be called.
+    :param env: environment variable mapping used for the prefetch.
+    :raises PackageManagerError: if the 'yarn install' command fails.
+    """
+    run_yarn_cmd(
```

Comment:
I believe it would be useful to add a `log.debug` that prints the exact command that's being invoked, similar to what we do with `gomod.Go._run`.

I know this is a shared function, so if you prefer, let's do it as a follow up.